syntax = "proto3";

package notification;

service NotificationService {
    rpc SendNotification(NotificationRequest) returns (NotificationResponse);
    rpc GetUserNotifications(UserNotificationsRequest) returns (UserNotificationsResponse);
}

message NotificationRequest {
    string user_id = 1;
    string type = 2; // ORDER_CREATED, PAYMENT_SUCCESS, PAYMENT_FAILED, ORDER_SHIPPED, etc.
    string title = 3;
    string message = 4;
    map<string, string> metadata = 5;
}

message UserNotificationsRequest {
    string user_id = 1;
    int32 limit = 2;
    int32 offset = 3;
}

message NotificationResponse {
    string notification_id = 1;
    string user_id = 2;
    string type = 3;
    string title = 4;
    string message = 5;
    bool sent = 6;
    string created_at = 7;
}

message UserNotificationsResponse {
    repeated NotificationResponse notifications = 1;
    int32 total = 2;
}